composer/callbacks/speed_monitor.py

```diff
@@ -108,11 +140,36 @@ def batch_end(self, state: State, logger: Logger):
         self.batch_wct_buffer.append(batch_wct)
         self.batch_num_samples_buffer.append(batch_num_samples)
 
-        # Log the throughput
         if len(self.batch_num_samples_buffer) == self.window_size:
+            # Log the throughput
             throughput = sum(self.batch_num_samples_buffer) / sum(self.batch_wct_buffer)
             logger.log_metrics({'throughput/samples_per_sec': throughput})
 
+            # Estimate remaining time
+            batch_wct_avg = sum(self.batch_wct_buffer) / len(self.batch_wct_buffer)
+            elapsed_dur = self.get_elapsed_duration(state)
+            if elapsed_dur is None:
+                warnings.warn('`max_duration` is not set. Cannot estimate remaining time.')
```

Comment:
if you don't set `max_duration`, I assume we just run one epoch of your dataset? And we don't know have any insight into how many samples are in your dataset? Is that correct?

Reply:
Correct. Its until dataloader is exhausted